independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/ListCategories.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.devtools.commands;
+
+import io.quarkus.devtools.commands.data.QuarkusCommandException;
+import io.quarkus.devtools.commands.data.QuarkusCommandInvocation;
+import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
+import io.quarkus.devtools.commands.handlers.ListCategoriesCommandHandler;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+import io.quarkus.devtools.project.QuarkusProject;
+import io.quarkus.platform.tools.ToolsConstants;
+import io.quarkus.platform.tools.ToolsUtils;
+import java.util.HashMap;
+
+/**
+ * Instances of this class are not thread-safe. They are created per single invocation.
+ */
+public class ListCategories {
+    public static final String NAME = "list-categories";
+    private static final String PARAM_PREFIX = ToolsUtils.dotJoin(ToolsConstants.QUARKUS, NAME);
+    public static final String FROM_CLI = ToolsUtils.dotJoin(PARAM_PREFIX, "fromCli");
+    public static final String FORMAT = ToolsUtils.dotJoin(PARAM_PREFIX, "format");
+    public static final String BATCH_MODE = ToolsUtils.dotJoin(PARAM_PREFIX, "batchMode");
+
```

Comment:
I would like to simplify the constants to flat strings. These values don't change, there isn't a strong reason to munge them through a tool first, IMO. We have to lookup and debug against these values, let's make them more obvious and pick a naming convention (e.g. same as quarkus application config conventions? or lower-snake-case? or lower-kebab-case?)

Reply:
I like e.g. `"quarkus.list-extensions.batch-mode"`, which I think is what Quarkus config uses. Is has nice structure in it.

Reply:
Done - you can check if you like the format.

Reply:
Now I can see the structure, and I can see what I should be looking for in maps too.